lib/charms/data_platform_libs/v0/data_interfaces.py

```diff
@@ -1078,3 +1105,160 @@ def _on_relation_changed_event(self, event: RelationChangedEvent) -> None:
                 event.relation, app=event.app, unit=event.unit
             )  # here check if this is the right design
             return
+
+
+# Opensearch related events
```

Comment:
Uhm... this should not really be here, but one should rather commit these modifications in the dataplatform-lib [here](https://github.com/canonical/data-platform-libs/blob/main/lib/charms/data_platform_libs/v0/data_interfaces.py). We should really never modify the libs within the projects but rather submit PR to the "upstream"

Although it may make sense to have custom events/classes, one thing that we were trying to promote was the use (whenever possible) of same classes across databases. I'm wondering whether one could use the already existing classes, just for the sake of avoiding to create too many granular classes. 

Also, classes there should follow the requirements that are outlined in https://github.com/canonical/charm-relation-interfaces

I understand the process there may be a bit convoluted (requiring other people to review the interface spec), but I suppose the idea is to try to make interfaces and data-platform libs somewhat more opinionated and controlled. 

If you want to get this merged, maybe it could be better to move these classes in the opensearch library for the time being, while starting the approval/review process for the interface

@delgod what is your point of view here?

Reply:
These changes were actually proposed by Mykola so we can have an interface that's more specific to opensearch; I'm aware that changes to this library should be proposed in the data-platform-lib repom, but I wanted to implement and test them in this repo first so we can ensure the library is fit for purpose before publishing :)